src/AppInstallerCommonCore/Fonts.cpp

```diff
@@ -0,0 +1,188 @@
+            THROW_HR_IF(E_OUTOFMEMORY, familyNameLength > ARRAYSIZE(familyName));
+            THROW_IF_FAILED(familyNames->GetString(0, &familyName[0], familyNameLength));
+            return std::wstring(familyName);
+        }
+    }
+
+    std::vector<FontFamily> GetInstalledFontFamilies()
+    {
+        wil::com_ptr<IDWriteFactory7> factory;
+        THROW_IF_FAILED(DWriteCreateFactory(DWRITE_FACTORY_TYPE_SHARED, __uuidof(factory), factory.put_unknown()));
+
+        wil::com_ptr<IDWriteFontCollection> collection;
+        THROW_IF_FAILED(factory->GetSystemFontCollection(collection.addressof(), FALSE));
+
+        const auto familyCount = collection->GetFontFamilyCount();
+
+        std::vector<FontFamily> fontFamilies;
+        for (UINT32 index = 0; index < familyCount; index++)
+        {
+            wil::com_ptr<IDWriteFontFamily> family;
+            THROW_IF_FAILED(collection->GetFontFamily(index, family.addressof()));
+
+            std::wstring familyName = GetFontFamilyName(family);
+
+            std::vector<FontFace> fontFaces;
+            const auto& fontCount = family->GetFontCount();
+            for (UINT32 fontIndex = 0; fontIndex < fontCount; fontIndex++)
+            {
+                wil::com_ptr<IDWriteFont> font;
+                THROW_IF_FAILED(family->GetFont(fontIndex, font.addressof()));
+                std::wstring faceName = GetFontFaceName(font);
+
+                wil::com_ptr<IDWriteFontFace> fontFace;
+                THROW_IF_FAILED(font->CreateFontFace(fontFace.addressof()));
+
+                const auto& filePaths = GetFontFilePaths(fontFace);
+
+                FontFace fontFaceEntry;
+                fontFaceEntry.Name = std::wstring(faceName);
+                fontFaceEntry.FilePaths = filePaths;
+                fontFaces.emplace_back(fontFaceEntry);
+            }
+
+            FontFamily fontFamily;
+            fontFamily.Name = familyName;
+            fontFamily.Faces = fontFaces;
+            fontFamilies.emplace_back(std::move(fontFamily));
+        }
+
+        return fontFamilies;
+    }
+
+    std::optional<FontFamily> GetInstalledFontFamily(const std::wstring& familyName)
+    {
+        wil::com_ptr<IDWriteFactory7> factory;
+        THROW_IF_FAILED(DWriteCreateFactory(DWRITE_FACTORY_TYPE_SHARED, __uuidof(factory), factory.put_unknown()));
+
+        wil::com_ptr<IDWriteFontCollection> collection;
+        THROW_IF_FAILED(factory->GetSystemFontCollection(collection.addressof(), FALSE));
+
+        UINT32 familyIndex;
+        BOOL familyExists;
+        THROW_IF_FAILED(collection->FindFamilyName(familyName.c_str(), &familyIndex, &familyExists));
+
+        if (!familyExists)
+        {
+            return {};
+        }
+
+        wil::com_ptr<IDWriteFontFamily> family;
+        THROW_IF_FAILED(collection->GetFontFamily(familyIndex, family.addressof()));
+        const auto fontCount = family->GetFontCount();
```

Comment:
Factor the "create a `FontFamily` from a `IDWriteFontFamily`" into a shared function.